packages/smooth_app/lib/main.dart

```diff
@@ -53,54 +53,44 @@ class _SmoothAppState extends State<SmoothApp> {
   late LocalDatabase _localDatabase;
   late ThemeProvider _themeProvider;
   bool systemDarkmodeOn = false;
-
-  Future<void> _init(BuildContext context) async {
-    _userPreferences = await UserPreferences.getUserPreferences();
-    _productPreferences = ProductPreferences(
-      ProductPreferencesSelection(
-        setImportance: (
-          String attributeId,
-          String importanceId,
-        ) async =>
-            _userPreferences.setImportance(attributeId, importanceId),
-        getImportance: (String attributeId) =>
-            _userPreferences.getImportance(attributeId),
-        notify: () => _productPreferences.notifyListeners(),
-      ),
-    );
-    try {
-      await _productPreferences.loadReferenceFromAssets(
-        DefaultAssetBundle.of(context),
-      );
-    } catch (e) {
-      // this is problematic - we should always be able to load the default
-      debugPrint('Could not load reference files: $e');
-      rethrow;
-    }
-    await _userPreferences.init(_productPreferences);
-    try {
-      _localDatabase = await LocalDatabase.getLocalDatabase();
-    } catch (e) {
-      // this is problematic - we should always be able to init the database
-      debugPrint('Cannot init database: $e');
-      rethrow;
-    }
-    _themeProvider = ThemeProvider(_userPreferences);
-  }
+  late Future<void> _initFuture;
 
   @override
   void initState() {
+    super.initState();
+    _initFuture = _init();
+  }
+
+  Future<void> _init() async {
+    Function debugPrintAndRethrow(String message) => (dynamic error) {
+          debugPrint('$message: $error');
+          return error;
```

Comment:
Maybe `'$message: $error' ` would make sense.

Reply:
Not sure what you mean. Note that this refactoring preserves the original error messages.

Reply:
This is what I had in mind: not even trying to go on with the app if there's a fatal error, and give the end-user enough data to complain about. In that test I've changed the name of the preference file. I send a PR today.
![Screenshot_2021-08-06-09-14-13](https://user-images.githubusercontent.com/11576431/128471516-c900988d-219f-4504-a60b-3cc3ba60365f.png)

Reply:
I don't think the end user can do anything with this information. I'd just have something that says "Open Food Facts has encountered a fatal error" with a "close" button.  You might have some small text with "additional information" that includes this string, to help with debugging.

Reply:
@gspencergoog That's correct: the end user won't be able to do anything, except see that obviously something very bad happened and potentially complain about it. That's part of the purpose of the PR: stop the app - which was not the case before.
Still, they will have a message that could help understand/debug, perhaps (probably not).
I did not want to put too much effort on a screen that hopefully no end user will see. Developers may see that screen once in a while, in debug phases.